homeassistant/components/ebox/sensor.py

```diff
@@ -96,9 +156,10 @@ def __init__(self, ebox_data, sensor_type, name):
         """Initialize the sensor."""
         self.client_name = name
         self.type = sensor_type
-        self._name = SENSOR_TYPES[sensor_type][0]
-        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
-        self._icon = SENSOR_TYPES[sensor_type][2]
+        metadata = SENSOR_TYPES[sensor_type]
+        self._name = metadata.name
+        self._attr_unit_of_measurement = metadata.unit_of_measurement
+        self._attr_icon = metadata.icon
         self.ebox_data = ebox_data
         self._state = None
```

Comment:
This can use `self._attr_state` in the update method.

Reply:
I would say that is a tiny bit out of scope, as the state has no relationship with the NamedTuple change this PR is about

Reply:
This is correct. That's why I just suggested as "can". But it's just a tiny addition after changing the other to using `_attr_`s.

Reply:
I prefer not to change the whole integration, that is not the purpose.